Tools/Scripts/generate-static-analysis-archive

```diff
@@ -58,30 +101,49 @@ def parse_command_line(args):
-            file_name = file_path.split('/')[-1].removeprefix(category).removesuffix('.txt')
-            relative_file_path = os.path.relpath(file_path, output_root)
-            with open(os.path.abspath(os.path.join(output_root, file_path))) as f:
-                count = len(f.read().splitlines())
-            if count:
-                checker_links += PROJECT_TEMPLATE.format(project_file_url=relative_file_path, project_name=file_name, project_issue_count=count)
-        if checker_links:
-            results_links += CATEGORY_TEMPLATE.format(category_name=category_formatted, checker_links=checker_links)
-
-    return results_links
+def generate_unexpected_table(output_root, static_analysis_dir, project_name, category):
+    # Get lists of unexpected results per project
+    category_table = ''
+    command = 'find {} -name {}\\* -print'.format(os.path.abspath(os.path.join(static_analysis_dir, project_name)), category)
+    try:
+        result_files = subprocess.check_output(command, shell=True, text=True)
+    except subprocess.CalledProcessError as e:
+        sys.stderr.write(f'{e.output}')
+        sys.stderr.write(f'Could not find results for {project_name}\n')
+        return -1
+    unexpected_results_files = result_files.splitlines()
+
+    # Add links to unexpected lists and counts in a table
+    table_rows = ''
+    for file_path in unexpected_results_files:
+        file_name = file_path.split('/')[-1].removeprefix(category).removesuffix('.txt')
+        relative_file_path = os.path.relpath(file_path, output_root)
+        with open(os.path.abspath(os.path.join(output_root, file_path))) as f:
+            count = len(f.read().splitlines())
+        if count:
+            table_rows += TABLE_ITEM_TEMPLATE.format(checker_type=file_name, link=LINK_TEMPLATE.format(link=relative_file_path, text=count))
+    if table_rows:
+        category_table += UNEXPECTED_TABLE_TEMPLATE.format(table_body_contents=table_rows)
+
+    return category_table
+
+
+def generate_unexpected_results_page(project_dirs, id_string, output_root, static_analysis_dir):
+    body = ''
+    for category, category_formatted in {'UnexpectedFailures': 'Unexpected Failures', 'UnexpectedPasses': 'Unexpected Passes'}.items():
+        body += f'<h2>{category_formatted}</h2>'
+        body += '<div class="inner">'
+        for project_name in sorted(project_dirs):
+            # For each project, link to all unexpected issues and create tables for failures and passes
+            static_analyzer_index_path = os.path.join(static_analysis_dir, project_name, INDEX_HTML)
+            project_count = get_project_issue_count_as_string(static_analyzer_index_path)
+            body += f'<h3>{project_name}</h3>'
+            if int(project_count) != 0 and category == 'UnexpectedFailures':
+                body += f"<b>{LINK_TEMPLATE.format(link=os.path.relpath(static_analyzer_index_path, output_root), text=f'View Issues ({project_count})')}</b></br></br>"
```

Comment:
As I commented in slack, we may want to omit this number since it's rather confusing to have two different kinds of numbers in the results page.

Reply:
Agreed